lib/platforms/identity.js

```diff
@@ -288,7 +288,7 @@ function findIdentitiesByFieldValue(field, values, callback) {
   Identity.find().where(field).in(values).exec(function(err, foundIdentities) {
 
     if (err) {
-      return callback(err);
+      return callback(new errors.InternalError('Could not find identities for field ' + field + ' and values ' + values));
```

Comment:
@guumaster  Perhaps the messages are incorrect. When a database error occurs it does not necessary mean that the requested object(s) were not found, but it means that something bad happened while attempting to manipulate them. The NotFoundError should apply when the returned results are empty or undefined.

Therefore, I can change the messages to something like `Fatal error while searching for Identity`, and keep the 500 status.